crates/bevy_ecs/src/entity/mod.rs

```diff
@@ -256,17 +261,41 @@ impl Entity {
     /// No particular structure is guaranteed for the returned bits.
     #[inline(always)]
     pub const fn to_bits(self) -> u64 {
-        (self.generation as u64) << 32 | self.index as u64
+        (self.generation.get() as u64) << 32 | self.index as u64
     }
 
     /// Reconstruct an `Entity` previously destructured with [`Entity::to_bits`].
     ///
     /// Only useful when applied to results from `to_bits` in the same instance of an application.
-    #[inline(always)]
+    ///
+    /// # Panics
+    ///
+    /// This method will likely panic if given `u64` values that did not come from [`Entity::to_bits`]
+    #[inline]
     pub const fn from_bits(bits: u64) -> Self {
```

Comment:
Perhaps 
```rust
#[inline(always)]
```
here? Minor nit, I'm not sure how hot this code path gets, ignore if non-hot.